packages/dcos-integration-test/extra/test_ucr.py

```diff
@@ -0,0 +1,208 @@
+
+    TODO(gilbert): add description.
+    """
+    test_uuid = uuid.uuid4().hex
+    app = {
+        'id': TEST_APP_NAME_FMT.format(test_uuid),
+        'cpus': 0.1,
+        'mem': 32,
+        'instances': 1,
+        'cmd': 'ls -al /',
+        'container': {
+            'type': 'MESOS',
+            'docker': {
+                'imgae': 'library/alpine@sha256:9f08005dff552038f0ad2f46b8e65ff3d25641747d3912e3ea8da6785046561a'
+            }
+        }
+    }
+    dcos_api_session.marathon.deploy_and_cleanup(app)
+
+
+def test_if_ucr_pods_can_be_deployed_with_image_entrypoint(dcos_api_session):
+    """Marathon pods inside ucr deployment integration test.
+
+    This test verifies that a Marathon pods can be deployed.
+    """
+    test_uuid = uuid.uuid4().hex
+    pod_definition = {
+        'id': '/integration-test-pods-{}'.format(test_uuid),
+        'scaling': {'kind': 'fixed', 'instances': 1},
+        'containers': [
+            {
+                'name': 'image_entrypoint',
+                'resources': {
+                    'cpus': 0.1,
+                    'mem': 32
+                },
+                'image': {
+                    'kind': 'DOCKER',
+                    'id': 'mesosphere/inky'
+                }
+            }
+        ],
+        'networks': [{'mode': 'host'}]
+    }
+    dcos_api_session.marathon.deploy_pod_and_cleanup(pod_definition)
+
+
+def test_if_ucr_pods_can_be_deployed_with_scratch_image(dcos_api_session):
+    """Marathon pods inside ucr deployment integration test.
+
+    This test verifies that a Marathon pods can be deployed.
+    """
+    test_uuid = uuid.uuid4().hex
+    pod_definition = {
+        'id': '/integration-test-pods-{}'.format(test_uuid),
+        'scaling': {'kind': 'fixed', 'instances': 1},
+        'containers': [
+            {
+                'name': 'image_entrypoint',
+                'resources': {
+                    'cpus': 0.1,
+                    'mem': 32
+                },
+                'image': {
+                    'kind': 'DOCKER',
+                    'id': 'hello-world'
+                }
+            }
+        ],
+        'networks': [{'mode': 'host'}]
+    }
+    dcos_api_session.marathon.deploy_pod_and_cleanup(pod_definition)
+
+
+def test_if_ucr_pods_can_be_deployed_with_image_whiteout(dcos_api_session):
+    """Marathon pods inside ucr deployment integration test.
+
+    This test verifies that a Marathon pods can be deployed.
+    """
+    test_uuid = uuid.uuid4().hex
+    pod_definition = {
+        'id': '/integration-test-pods-{}'.format(test_uuid),
+        'scaling': {'kind': 'fixed', 'instances': 1},
+        'containers': [
+            {
+                'name': 'image_entrypoint',
+                'resources': {
+                    'cpus': 0.1,
+                    'mem': 32
+                },
+                'image': {
+                    'kind': 'DOCKER',
+                    'id': 'gilbertsong/whiteout'
```

Comment:
Can we upload this image to a mesosphere account and do a pull from there. Would try to avoid pulling images from personal accounts.